### PR TITLE
MR-CANHUBK3 ADAP board support, add ADC support

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
Adds support for the MR-CANHUBK3-ADAP adapter board for [MR-CANHUBK344](https://www.nxp.com/design/development-boards/automotive-development-platforms/s32k-mcu-platforms/s32k344-evaluation-board-for-mobile-robotics-incorporating-100baset1-and-six-can-fd:MR-CANHUBK344) that includes

- SD Card slot
- 10-pin JST-GH GPS connector
- Safety button
- VBAT ADC input
- Dedicated RC connector
- IMC42688P IMU
- BMM150 Magnetometer

Detection is done on runtime so that the MR-CANHUBK344 board can run the same firmware.

### CODE REVIEW

**Comment by PetervdPerk-NXP** [comment]
Re-based on main branch.
CI looking better I think it's ready to merge

File: boards/nxp/mr-canhubk3/nuttx-config/include/board.h [hunk 1]
```diff
@@ -152,6 +152,18 @@
 #define PIN_LPUART1_RX    (PIN_LPUART1_RX_3 | PIN_INPUT_PULLUP)   /* PTC6 */
 #define PIN_LPUART1_TX    PIN_LPUART1_TX_3   /* PTC7 */
 
+
+/* LPUART4 /dev/ttyS3  P8B 3X7 Pin 3 Single wire RC UART */
+
+#define PIN_LPUART4_RX    0                  /* NC */
```
Comment (davids5) [inline boards/nxp/mr-canhubk3/nuttx-config/include/board.h:158]: This can cause problems.

File: platforms/nuttx/CMakeLists.txt [hunk 2]
```diff
@@ -383,6 +383,9 @@ if(NOT NUTTX_DIR MATCHES "external")
 	if(CONFIG_ARCH_CHIP_MIMXRT1062DVL6A)
 		set(DEBUG_DEVICE "MIMXRT1062XXX6A")
 		set(DEBUG_SVD_FILE "MIMXRT1052.svd")
+	elseif(CONFIG_ARCH_CHIP_MIMXRT1176DVMAA)
```
Comment (davids5) [inline platforms/nuttx/CMakeLists.txt:386]: This is unrelated. Remove?

File: platforms/nuttx/src/px4/nxp/s32k3xx/adc/adc.cpp [hunk 3]
```diff
@@ -186,10 +118,10 @@ float px4_arch_adc_reference_v()
 
 uint32_t px4_arch_adc_temp_sensor_mask()
 {
-	return 1 << (ADC_SC1_ADCH_TEMP >> ADC_SC1_ADCH_SHIFT);
+	return 1 << 0; // No temp sensor
```
Comment (davids5) [inline platforms/nuttx/src/px4/nxp/s32k3xx/adc/adc.cpp:121]: Return 0

**Review by davids5 — CHANGES_REQUESTED** [review]
@PetervdPerk-NXP Looks good except for as noted.

File: boards/nxp/mr-canhubk3/nuttx-config/include/board.h (hunk 1 above)
Comment (PetervdPerk-NXP) [reply boards/nxp/mr-canhubk3/nuttx-config/include/board.h:158]: I simply don't have a LPUART4_RX pin, but that's isn't a problem for the RC UART since the TX pin is bi-directional.

Comment (davids5) [reply boards/nxp/mr-canhubk3/nuttx-config/include/board.h:158]: I understand that, but the question is
What port/pin will be manipulated in the open when  `s32k3xx_pinconfig(PIN_LPUART4_RX)`  is called with 0? https://github.com/PX4/NuttX/blob/px4_firmware_nuttx-10.3.0%2B/arch/arm/src/s32k3xx/s32k3xx_lowputc.c#L267

Comment (PetervdPerk-NXP) [reply boards/nxp/mr-canhubk3/nuttx-config/include/board.h]: You're right. I think the following change would work better.
https://github.com/PX4/PX4-Autopilot/blob/23ba37febebefcdef1ec3d778f5b139f606b7fef/boards/nxp/mr-canhubk3/nuttx-config/include/board.h#L158

Comment (davids5) [reply boards/nxp/mr-canhubk3/nuttx-config/include/board.h]: Yes I thought of that as well, but it is is a tad bit yuck! Could do -1, and remove the debug asserts.

Comment (PetervdPerk-NXP) [reply boards/nxp/mr-canhubk3/nuttx-config/include/board.h]: We're using all 32 bits of the UINT32_T so might have yield bad consequences then

Comment (davids5) [reply boards/nxp/mr-canhubk3/nuttx-config/include/board.h]: I was being lazy with -1, use UINT32_MAX it iwll pard out to port too high and pin to high

Comment (davids5) [reply boards/nxp/mr-canhubk3/nuttx-config/include/board.h]: Then can this even happen?
```
 if ((port >= S32K3XX_NPORTS) && (pin >= S32K3XX_NPINS))
    {
      return -EINVAL; /* Invalid port or pin number */
    }
```

Comment (PetervdPerk-NXP) [reply boards/nxp/mr-canhubk3/nuttx-config/include/board.h]: Nope that can't happen 😅

**Comment by hamishwillee** [comment]
FYI @PetervdPerk-NXP the only NXP board that the user guide suggests is available is http://docs.px4.io/main/en/flight_controller/nxp_rddrone_fmuk66.html#nxp-rddrone-fmuk66-fmu. I hope there is a plan to add information on other boards too.

File: boards/nxp/mr-canhubk3/nuttx-config/include/board.h [hunk 4]
```diff
@@ -152,6 +152,18 @@
 #define PIN_LPUART1_RX    (PIN_LPUART1_RX_3 | PIN_INPUT_PULLUP)   /* PTC6 */
 #define PIN_LPUART1_TX    PIN_LPUART1_TX_3   /* PTC7 */
 
+
+/* LPUART4 /dev/ttyS3  P8B 3X7 Pin 3 Single wire RC UART */
+
+#define PIN_LPUART4_RX    PIN_LPUART4_TX_3   /* PTE11 */
```
Comment (davids5) [inline boards/nxp/mr-canhubk3/nuttx-config/include/board.h:158]: Let's Add a comment describing why this done this way and is not a typo

**Review by davids5 — CHANGES_REQUESTED** [review]

File: boards/nxp/mr-canhubk3/nuttx-config/include/board.h (hunk 4 above)
Comment (PetervdPerk-NXP) [reply boards/nxp/mr-canhubk3/nuttx-config/include/board.h:158]: Done

**Review by davids5 — APPROVED** [review]
